### PR TITLE
post_live_metrics: Add `_post_in_chunks`.

### PR DESCRIPTION
**Pull request by daavoo**
- Post metrics and parameters separately
- Try to post as many plots as possible and stop before reaching 30MB.
- Discard plots higher than 10MB.

Optimistically? expect metrics and parameters to not exceed 30MB.

### CODE REVIEW

**Comment by dberenbaum** [comment]
Any concerns about this spamming studio too much? Thinking of https://iterativeai.slack.com/archives/C01116VLT7D/p1692452930044229.

**Comment by daavoo** [comment]
> Any concerns about this spamming studio too much? Thinking of https://iterativeai.slack.com/archives/C01116VLT7D/p1692452930044229.

As I understood it, the problem in the link was not about too many requests from the client to the server.
 The problem was (still is) in the backend querying the database in a very inefficient way, so a single request in a project with a lot of steps per linear plot caused / can trigger this (https://github.com/iterative/studio/issues/7256).

Anyhow, the change here mainly affects images, which are handled differently from linear plots (the source of the linked problem).

The main difference is when there the size of the images exceeds the server limit per single request:

Without this P.R, we just fail to send data on each step and try to send it in the next one. Failing forever and the request keeps getting bigger.
With this P.R:, we are able to send the data at each step bypassing the server limit.

**Comment by shcheklein** [comment]
I think it doesn't have to wait. The only thing that comes to my mind to check (besides the basic review, @amritghimire can you do that please?)- align the limits with Studio `big_files` check. See here for example - https://github.com/iterative/studio/pull/7299/files#diff-a67d283c73e8021405df0b1de3fb8dcf570563c21c70de5324cc2d8b661e561fR36

We skip 10Mb files on Studio (so they will disappear as we then parse an experiment).

Also, I now put a limit of 200 files per directory (not ideal, see the discussion and chime in the PR)- but as a safe guard. Here (or on the Studio side? or both?) we probably also want to limit the number of chunks, or number of files, etc.

File: src/dvc_studio_client/post_live_metrics.py [hunk 1]
```diff
@@ -12,6 +14,8 @@
 from .env import DVC_STUDIO_TOKEN, STUDIO_ENDPOINT, STUDIO_TOKEN
 from .schema import SCHEMAS_BY_TYPE
 
+MAX_CHUNK_SIZE = 29000000
```
Comment (amritghimire) [inline src/dvc_studio_client/post_live_metrics.py:17]: Lets add a comment with human readable size for an easy picture here,

**Review by amritghimire — CHANGES_REQUESTED** [review]

**Comment by daavoo** [comment]
> I think it doesn't have to wait. The only thing that comes to my mind to check (besides the basic review, @amritghimire can you do that please?)- align the limits with Studio `big_files` check. See here for example - https://github.com/iterative/studio/pull/7299/files#diff-a67d283c73e8021405df0b1de3fb8dcf570563c21c70de5324cc2d8b661e561fR36
> 
> We skip 10Mb files on Studio (so they will disappear as we then parse an experiment).

> Also, I now put a limit of 200 files per directory (not ideal, see the discussion and chime in the PR)- but as a safe guard. Here (or on the Studio side? or both?) we probably also want to limit the number of chunks, or number of files, etc.

Added both checks

**Pull request merged**